example-project/README.md

```diff
@@ -0,0 +1 @@
+# Example project
```

Comment:
What's the purpose of this example-project directory? To show how to use Typescript or was this a temporary local testing directory?

If it's intended to stick around as an example, can we instead move this to the `examples` directory and call it `typescript` and then elaborate some starter details in this README?

Reply:
> Is this PR ready for final review? Looks like everything has been split out into separate files at this point?

- Yes, all the types have been reorganized. I also made some minor additions/fixes. Ready for final review!

___

> What's the purpose of this example-project directory? To show how to use Typescript or was this a temporary local testing directory?
> 
> If it's intended to stick around as an example, can we instead move this to the `examples` directory and call it `typescript` and then elaborate some starter details in this README?

- Just a temporary directory for testing implementation in typescript. 
- Can be used during review to verify types are correct.

We can probably go ahead and remove this directory after final review.

Reply:
I spoke too soon - sorry! I forgot I still have some resources to add to the API class.

A review could still be helpful at this point.  Any input your team might have would be much appreciated!

Thanks!